app/main.py

```diff
@@ -1,3 +1,77 @@
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: (float, int)) -> None:
+        self.km = km
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __add__(self, other: (object, float, int)) -> object:
+        if isinstance(other, (float, int)):
+            create_object_from_int = Distance(other)
+            return Distance(self.km + create_object_from_int.km)
+
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other: (object, float, int)) -> object:
+        if isinstance(other, (float, int)):
+            create_object_from_int = Distance(other)
+            self.km = self.__add__(create_object_from_int).km
+
+            return self
+
+        self.km = self.__add__(other)
+        return self
+
+    def __mul__(self, other: (object, float, int)) -> object:
+        if isinstance(other, (float, int)):
+            create_object_from_int = Distance(other)
+            return Distance(self.km * create_object_from_int.km)
+
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: object) -> object:
+
+        return Distance(round(self.km / other, 2))
```

Comment:
The `__truediv__` method assumes `other` is always a number, but it should also handle the case where `other` is a `Distance` object. You need to check the type of `other` and handle division accordingly.